service/configinit_test.go

```diff
@@ -0,0 +1,275 @@
+		ballastextension.NewFactory(),
+		zpagesextension.NewFactory(),
+	)
+
+	factories.Receivers, _ = component.MakeReceiverFactoryMap(
+		otlpreceiver.NewFactory(),
+	)
+
+	factories.Exporters, _ = component.MakeExporterFactoryMap(
+		loggingexporter.NewFactory(),
+		otlpexporter.NewFactory(),
+		otlphttpexporter.NewFactory(),
+	)
+
+	factories.Processors, _ = component.MakeProcessorFactoryMap(
+		batchprocessor.NewFactory(),
+		memorylimiterprocessor.NewFactory(),
+	)
+
+	return factories
+}
+
+func cleanup() {
+	exportersSelected, processorsSelected, receiversSelected, extensionsSelected, pipelinesSelected = []string{}, []string{}, []string{}, []string{}, []string{}
+}
+func TestNewConfigurationInitCommand(t *testing.T) {
+	t.Cleanup(cleanup)
+	factories := components()
+	tests := []struct {
+		name    string
+		args    []string
+		wantErr bool
+	}{
+		{
+			name: "valid",
+			args: []string{"-p", "batch", "-m", "./../", "-d"},
+		},
+		{
+			name:    "invalid-no-module",
+			args:    []string{},
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			cmd := NewConfigurationInitCommand(CollectorSettings{Factories: factories})
+			cmd.SilenceUsage = true
+			origCfgCmd := cfgCmd
+			cfgCmd.exiter = func(args ...interface{}) error {
+				return fmt.Errorf("%v", args...)
+			}
+			cmd.SetArgs(tt.args)
+			err := cmd.Execute()
+			if tt.wantErr {
+				assert.Error(t, err)
+			}
+			cfgCmd = origCfgCmd
+		})
+	}
+}
+
+func TestGetComponentNodeErrors(t *testing.T) {
+	t.Cleanup(cleanup)
+	factories := components()
+	tests := []struct {
+		name string
+		args []string
+	}{
+		{
+			name: "invalid-exporter",
+			args: []string{"-e", "noexp", "-m", "./../"},
+		},
+		{
+			name: "invalid-processor",
+			args: []string{"-p", "nopro", "-m", "./../"},
+		},
+		{
+			name: "invalid-receiver",
+			args: []string{"-r", "norec", "-d", "-m", "./.."},
+		},
+		{
+			name: "invalid-extension",
+			args: []string{"-x", "noext", "-d", "-m", "./.."},
+		},
+		{
+			name: "invalid-service",
+			args: []string{"-l", "pipeline-fail", "-m", "./.."},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			cmd := NewConfigurationInitCommand(CollectorSettings{Factories: factories})
+			cmd.SilenceUsage = true
+			origCfgCmd := cfgCmd
+			cfgCmd.exiter = func(args ...interface{}) error {
+				return fmt.Errorf("%v", args...)
+			}
+			cmd.SetArgs(tt.args)
+			err := cmd.Execute()
+			assert.Error(t, err)
+			cfgCmd = origCfgCmd
+		})
+	}
+}
+
+func TestSampleConfigFileCreation(t *testing.T) {
+	t.Cleanup(cleanup)
+	dir := t.TempDir()
+	file := fmt.Sprintf("%v/%v", dir, "otelcol_temp_config.yaml")
+	factories := components()
+	cmd := NewConfigurationInitCommand(CollectorSettings{Factories: factories})
+	cmd.SetArgs([]string{"-p", "batch", "-r", "otlp", "-m", "./..", "-f", file})
+	_ = cmd.Execute()
+
+	_, err := os.Stat(file)
+	if err != nil {
+		assert.Fail(t, "File %v was expected to be created", file)
```

Comment:
Why not just an `assert.NoError(t, err, "file ... was expected ...")` ?

Reply:
Yes. Thats much better